src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,156 @@
+            size++;
+        } else {
+            Node<T> insertingNode;
+            Node newNode = getNodeNyIndex(index - 1);
+            insertingNode = new Node<>(newNode, value, newNode.nextNode);
+            Node afteradditingNode = newNode.nextNode;
+            newNode.nextNode = insertingNode;
+            insertingNode.previousNode = newNode;
+            afteradditingNode.previousNode = insertingNode;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int j = 0; j < list.size(); j++) {
+            add(list.get(j));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        Node currentNote = getNodeNyIndex(index);
+        return (T) currentNote.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        Node<T> currentNode = getNodeNyIndex(index);
+        T valueReplacing = currentNode.value;
+        currentNode.value = value;
+        return valueReplacing;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> currentNode = getNodeNyIndex(index);
+        T removedValue = (T) currentNode.value;
+        delete(currentNode);
+        size--;
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = new Node<>(null, null, head);
+        for (int i = 0; i < size; i++) {
+            currentNode = currentNode.nextNode;
+            if (object == currentNode.value || object != null && object.equals(currentNode.value)) {
+                delete(currentNode);
+                size--;
+                return true;
+
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
+
+    private void indexCheck(int index) {
+        if (size <= index || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Index is out of bound");
+        }
+    }
+
+    private void indexCheckForAdd(int index) {
+        if (size < index || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Index is out of bound");
+        }
+    }
+
+    private Node getNodeNyIndex(int index) {
+        Node currentNode = null;
+        if (size / 2 - 1 < index) {
+            currentNode = tail;
+            for (int i = size - 1; i > index; i--) {
+                currentNode = currentNode.previousNode;
+            }
+        } else {
+            currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.nextNode;
+            }
+        }
+        return currentNode;
+    }
+
+    private void delete(Node removedNode) {
```

Comment:
this method is not correct. Use this unlink method:
```
private T delete(Node<T> removedNode) {
        T oldValue = removedNode.value;
        Node<T> prev = removedNode.previousNode;
        Node<T> next = removedNode.nextNode;

        if (prev == null) {
            head = next;
        } else {
            prev.nextNode = next;
        }
        if (next == null) {
            tail = prev;
        } else {
            next.previousNode = prev;
        }
        size--;
        return oldValue;
    }
```

Reply:
done